### PR TITLE
[!!!][FEATURE] Use DTO for record representation and identity

### PR DESCRIPTION
**Pull request by mabolek**
This internal change does not affect any external APIs, but improves the way record data and identities are handled. This fixes and issue that made multilingual operations impossible, as well as making it easier to work with record data and identities.

* `\Pixelant\Interest\Domain\Model\Dto\RecordRepresentation` represents a record state, and includes data as well as a `RecordInstanceIdentifier` object.
* `\Pixelant\Interest\Domain\Model\Dto\RecordInstanceIdentifier` represents a record's identity: table, UID, language, workspace, and remote ID. It can output both the remote ID as it is seen from the outside (which is only unique when combined with language and workspace) and the internal, so-called "aspected", remote ID, which is unique for each language and workspace combination.

### CODE REVIEW

File: Classes/DataHandling/Operation/AbstractRecordOperation.php [hunk 1]
```diff
@@ -313,6 +311,10 @@ private function resolveStoragePid(): int
             $settings['persistence.']['storagePid.'] ?? []
         );
 
+        if ($pid === '') {
```
Comment (MattiasNilsson) [inline Classes/DataHandling/Operation/AbstractRecordOperation.php:314]: Should this also cover the null check not only empty string?

File: Classes/Domain/Model/Dto/RecordInstanceIdentifier.php [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    public function __construct(
+        string $table,
+        string $remoteId,
+        ?string $language = null,
+        ?string $workspace = null
+    ) {
+        $this->table = strtolower($table);
+        $this->remoteId = $remoteId;
+        $this->workspace = $workspace;
+        $this->language = $this->resolveLanguage((string)$language);
```
Comment (MattiasNilsson) [inline Classes/Domain/Model/Dto/RecordInstanceIdentifier.php:59]: Is there a need to typecast this as it already is a string?

**Review by MattiasNilsson — COMMENTED** [review]
Really nice one, make the data structure more readable! I have some small comments.

File: Classes/Domain/Model/Dto/RecordInstanceIdentifier.php (hunk 2 above)
Comment (mabolek) [reply Classes/Domain/Model/Dto/RecordInstanceIdentifier.php:59]: It's a string *or* null, so by casting null to string we get an empty string. The question would be if this conversion should happen earlier.

File: Classes/DataHandling/Operation/AbstractRecordOperation.php (hunk 1 above)
Comment (mabolek) [reply Classes/DataHandling/Operation/AbstractRecordOperation.php:314]: Exchanging this condition with `empty($pid)`.

File: Classes/Domain/Model/Dto/RecordInstanceIdentifier.php (hunk 2 above)
Comment (mabolek) [reply Classes/Domain/Model/Dto/RecordInstanceIdentifier.php]: I have changed this now.

**Review by MattiasNilsson — APPROVED** [review]
Nice one 👍

**Pull request merged**